include/private/fnft__misc.h

```diff
@@ -235,6 +236,66 @@ FNFT_UINT fnft__misc_nextpowerof2(const FNFT_UINT number);
 FNFT_INT fnft__misc_resample(const FNFT_UINT D, const FNFT_REAL eps_t, FNFT_COMPLEX const * const q,
     const FNFT_REAL delta, FNFT_COMPLEX *const q_new);
 
+/**
+ * @brief Multiples two square matrices of size N.
+ *
+ * @ingroup  misc
+ *
+ * Multiples two square matrices U and T of size N. T is replaced by the
+ * result U*T.
+ * @param[in] N Positive integer that is the size of the two matrices.
+ * @param[in] U Pointer to the first element of complex values NxN matrix U.
+ * @param[in,out] T Pointer to the first element of complex values NxN matrix T.
+ * Contains the result U*T on return.
+ */
+static inline void fnft__misc_square_matrix_mult(const FNFT_UINT N, FNFT_COMPLEX * const U,
+        FNFT_COMPLEX *const T){
+    FNFT_UINT  c1, c2, c3;
+    FNFT_COMPLEX TM[32] = { 0 }, sum = 0;
```

Comment:
The size of `TM` is hard-coded, which I guess is for efficiency? For too large `N`, this might lead to undetected memory violations.

I would suggest to either add a check that `N` is not too large and mention the limit in the docs, or, if that really has an impact performance and it fits the further use in FNFT, define one prototype function

 `misc_square_matrix_mult_proto(N, U, T, TM)` 

and then introduced functions like 

`misc_matrix_mult_2x2(N, U, T)`, 
`misc_matrix_mult_3x3(N, U, T)` 

that locally define `TM` of the right size and pass it to the prototype.

Reply:
Changed to this suggested structure.